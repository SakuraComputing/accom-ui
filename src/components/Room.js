import React from 'react';
import ReactHtmlParser from 'react-html-parser';
import facilityList from '../common/facilityList';
import Availability from '../../data/accommodation_availability_data';

export class Room extends React.Component {

    constructor(props){
        super(props);
    }

    getRoomsAvailable() {
        let availability = Availability["rooms"].find(room => room.id === this.props.id);
        if(availability) {
            const { available }  = availability;
            return <span className="rooms-available">Rooms available: {available}</span>
        } else {
            return <div className="room-sold-out">Sold Out</div>;
        }
    }

    getprices () {
        const { supplement_price }   = this.props;

        if(supplement_price) {
            const { price: { price }} = supplement_price;
            return <div className="price">
                <span>Today's Price: { supplement_price ? supplement_price.price.price : '' }</span>
            </div>
        } else {
            return <div className="room-sold-out">You missed it our last room sold out</div>;
        }
    }

    displayReserveButton() {
        let availability = Availability["rooms"].find(room => room.id === this.props.id);
        if(availability) {
            return <div className="clearfix">
                <button className="button button--reserve button--div">I'll Reserve</button>
            </div>
        } else {
            return '';
        }
    }

    render() {

        return (
            <div className="room-container">
                <div className="details-container">
                    <div className="details-container__row">
                        <div className="details-container__col">
                            <div className="details-container__col__header">Room Name</div>
                            {this.props.name}
                        </div>
                        <div className="details-container__col">
                            <div className="details-container__col__header">Room Type</div>
                            {this.props.room_type_description}
                        </div>
                        <div className="details-container__col">
                            <div className="details-container__col__header">Occupancy</div>
                            <li><span>{this.props.occupancy} pers. </span>(Minimum: {this.props.min_occupancy})</li>
                        </div>
                        <div className="details-container__col">
                            { this.getRoomsAvailable() }
                        </div>
                        <div className="details-container__col">
                            { this.getprices()}
                        </div>
                    </div>
                </div>
                <div className="list-header">
                    <div>Facilities</div>
                    { this.props.facilities ? facilityList(this.props.facilities) : <li>Only Basic Facilities Avaliable</li>}
                </div>
                <div className="room-desc">
                    *{ReactHtmlParser(this.props.description)}
                </div>
                {this.displayReserveButton()}
            </div>
        )
    }
};

export default Room;