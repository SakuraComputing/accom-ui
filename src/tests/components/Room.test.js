import React from 'react';
import { shallow } from 'enzyme';
import Room from '../../components/Room';
import toJSON from 'enzyme-to-json';
import roomData from '../fixtures/rooms';

describe('Room Page', () => {

    let wrapper;

    it('should render the form correctly', () => {
        const id = 161;
        wrapper = shallow(<Room id={id}/>);
        expect(wrapper).toMatchSnapshot();
    });
    describe('Available Rooms', () => {
        it('should return span tag from getRoomsAvailable', () => {
            const available = 6;
            const returnRooms = wrapper.instance().getRoomsAvailable(available);
            const test = <span className="rooms-available">Rooms available: {available}</span>;
            expect(returnRooms).toEqual(test);
        });

        it('should return span tag with no rooms available', () => {
            const id = 16;
            wrapper = shallow(<Room id={id}/>);
            const returnNoRooms = wrapper.instance().getRoomsAvailable();
            const test = <th className="room-sold-out">Sold Out</th>;
            expect(returnNoRooms).toEqual(test);
        });
    });

    describe('Prices', () => {

        let id;

        beforeEach(() => {
            id = 136;
        });

        it('should return prices tag', () => {
            const supplement_price = { price: { price: "Â£409"} };
            wrapper = shallow(<Room id={id} supplement_price={supplement_price}/>);
            const priceField = wrapper.instance().getprices();
            const priceReturn = <div className="price">
                <span>Today's Price: { supplement_price ? supplement_price.price.price : '' }</span>
            </div>;
            expect(priceField).toEqual(priceReturn);
        });
        it('should not render the prices tag', () => {
            const supplement_price = { price: {} };
            wrapper = shallow(<Room id={id} supplement_price={supplement_price}/>);
            const priceField = wrapper.instance().getprices();
            const priceReturn = <div className="price">
                <span>Today's Price: </span>
            </div>;
            expect(toJSON(priceField)).toEqual(toJSON(priceReturn));
        })
    });
});
