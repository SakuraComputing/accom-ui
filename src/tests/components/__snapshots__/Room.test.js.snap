// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room Page should render the form correctly 1`] = `
<div
  className="room-container"
>
  <div
    className="details-container"
  >
    <div
      className="details-container__row"
    >
      <div
        className="details-container__col"
      >
        <th
          className="details-container__col__header"
        >
          Room Name
        </th>
      </div>
      <div
        className="details-container__col"
      >
        <th
          className="details-container__col__header"
        >
          Room Type
        </th>
      </div>
      <div
        className="details-container__col"
      >
        <th
          className="details-container__col__header"
        >
          Occupancy
        </th>
        <li>
          <span>
             pers. 
          </span>
          (Minimum: 
          )
        </li>
      </div>
      <div
        className="details-container__col"
      >
        <th
          className="room-sold-out"
        >
          Sold Out
        </th>
      </div>
      <div
        className="details-container__col"
      >
        <th
          className="room-sold-out"
        >
          You missed it our last room sold out
        </th>
      </div>
    </div>
  </div>
  <div
    className="list-header"
  >
    <th>
      Facilities
    </th>
    <li>
      Only Basic Facilities Avaliable
    </li>
  </div>
  <div
    className="room-desc"
  >
    *
  </div>
</div>
`;
